# Background
A recent [study](http://www.pnas.org/content/112/34/E4752) based on laboratory animals has reported polymorphism in the sex chromosomes of the frog *X. tropicalis*.  Because this study is based on lab animals of unknown provenance, it is not clear whether this variation in sex chromosomes naturally occurs within one population or whether it occurs in a geographically structured fashion

# Goals
To evaluate whether there is geographic structure to the mechanism of sex determination in the frog *X. tropicalis*

# Work to be done
This project has not yet been started.  It will involve fieldwork in Ghana which is scheduled for mid-year 2016.  On this expedition, we hopefully can collect *X. tropicalis* from multiple localities and export live aniamls as well as preserved animals. These animals could then be crossed in the lab, raised to postmetamorphosis, and screened for ZW versus XY sex systems using GBS.

# Anticipated Impact
I anticipate that this work will result in a medium to high impact publication suitable for a journal such as Evolution or Genome Biology and Evolution.  *Xenopus tropicalis* is a widely used organism for research and the genetics behind sex determination would thus be of keen interest to many.  This project would expose Caroline to fieldwork in Africa, and GBS bioinformatic approaches using a reference genome.

# Key collaborators
This would be a collaborative effort in the Evans lab with possible outside contributors, possibly led by Caroline.  This project cannot begin until fieldwork is completed in mid 2016, so other projects will be started in the meantime.

# Update from June, 2020
With Caroline and BenF, we did fieldwork in Ghana and brought back frogs which turned out to have Y chromosomes.  Now we are revising a paper and need to assess evidence for degeneration of the W-chromosome and also the potential for reference bias in the detection of male biased transcripts.  We are using three approaches for this:
* If Y is derived from Z and females have degeneration of the W, then males would be expected to have higher expression. 
* If reference bias is an issue - level of female expression should be lower than the autosomal. We can compare the median expression of SL and non-SL transcripts in females. Maybe get the ratio of these expression levels by individual.
* Map RNAseq data to v10. Count SNP/bps of non-male-specific male-biased transcripts in SL region and compare to SNPs/bp of non-male biased transcripts in SL region.
